lib/src/message/mod.rs

```diff
@@ -7,6 +7,61 @@
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 //
 
+use error::*;
+use error::ErrorKind as EK;
+use iter::{StripWhiteSpace, StripWhiteSpaceRightIter};
+use iter::{WithoutComments, WithoutCommentsIter};
+
 pub mod line;
 pub mod trailer;
 
+
+/// Special iterator extension for messages
+///
+/// This iterator extension provides some special functionality for issue and
+/// commit messages. It is intended for use on iterators over the lines of a
+/// message.
+///
+pub trait LineIteratorExt {
+    type Iter : Iterator<Item = String>;
+
+    /// Check whether the formatting of a message is valid
+    ///
+    /// This function checks whether a message has a subject line and whether
+    /// that subject line is followed by an empty line. The message should
+    /// already be stripped of comments and trailing whitespace.
+    ///
+    fn check_message_format(self) -> Result<()>;
+
+    /// Create a whitespace and comment stripping iterator
+    ///
+    /// This function creates an iterator suitable for stripping parts of a
+    /// message which should not be stored, e.g. commetents and trailing
+    /// whitespace.
+    ///
+    /// Note that the iterator does not (yet) strip blank lines at the beginning
+    /// or end of a message.
+    ///
+    fn stripped(self) -> StripWhiteSpaceRightIter<WithoutCommentsIter<Self::Iter>>;
+}
+
+impl<L> LineIteratorExt for L where L: Iterator<Item = String> {
+    type Iter = L;
+
+    fn check_message_format(mut self) -> Result<()> {
+        if try!(self.next().ok_or(Error::from_kind(EK::EmptyMessage))).is_empty() {
+            return Err(Error::from_kind(EK::EmptySubject))
+        }
+
+        if !self.next().map(|line| line.is_empty()).unwrap_or(true) {
+            return Err(Error::from_kind(EK::MalformedMessage));
+        }
+
+        Ok(())
+    }
+
+    fn stripped(self) -> StripWhiteSpaceRightIter<WithoutCommentsIter<Self::Iter>> {
```

Comment:
Yeah, and I even did not include the `SkipWhile` which I intended originally. I'll guess we have to wait until we can simply return an `Iterator<Item = String>`.